docs/modules/record-encryption/con-record-encryption-overview.adoc

```diff
@@ -73,7 +73,15 @@ It is recommended to use the KMS in a high availability (HA) configuration.
 Key rotation involves periodically replacing cryptographic keys with new ones and is considered a best practice in cryptography.
 
 The filter allows the rotation of Key Encryption Keys (KEKs) within the Key Management System (KMS). 
-When a KEK is rotated, the new key material is used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
+When a KEK is rotated, the new key material is eventually used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
```

Comment:
```suggestion
When a KEK is rotated, the new key material is eventually used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still available in the KMS.
```

Some of the KMS systems have a rather sophisticated view of the status of a KEK so it might still be in there but not accessible (at least thats how I understand the model)